File: ios/Releaf/Data/Notebook/BookSeriesEntity.swift
/*
 * BookSeriesEntity.swift
 *
 * GRDB record for the `book_series` table. A series groups notebook
 * rows (volumes) under a shared book name. Single-volume books
 * leave `series_id` NULL on the notebook row — no series row is
 * required until a second volume is added.
 *
 * Introduced in GRDB migration v3.
 */

import Foundation
import GRDB

public struct BookSeriesEntity: Codable, FetchableRecord, PersistableRecord,
                                 Identifiable, Equatable, Sendable {

    public static let databaseTableName = "book_series"

    public var id: String
    public var shelfId: String
    public var name: String
    public var createdAt: String
    public var updatedAt: String
    public var dirty: Bool
    public var deletedAt: String?

    public enum CodingKeys: String, CodingKey {
        case id
        case shelfId   = "shelf_id"
        case name
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case dirty
        case deletedAt = "deleted_at"
    }

    public init(
        id: String,
        shelfId: String,
        name: String,
        createdAt: String,
        updatedAt: String,
        dirty: Bool = true,
        deletedAt: String? = nil
    ) {
        self.id = id
        self.shelfId = shelfId
        self.name = name
        self.createdAt = createdAt
        self.updatedAt = updatedAt
        self.dirty = dirty
        self.deletedAt = deletedAt
    }
}

File: apps/releaf/android/app/src/main/java/app/releaf/mobile/data/notebook/BookSeriesEntity.kt
/*
 * BookSeriesEntity.kt
 *
 * Groups notebooks that are volumes of the same book. A series
 * belongs to exactly one shelf; a book row (volume) points back at
 * its series via `series_id`. Single-volume books carry
 * `series_id = NULL` so the UI can render them without a "Vol 1"
 * suffix.
 *
 * Introduced in DB v12.
 */

package app.releaf.mobile.data.notebook

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "book_series",
    indices = [
        Index("shelf_id"),
        Index("deleted_at"),
    ],
)
data class BookSeriesEntity(
    @PrimaryKey
    @ColumnInfo(name = "id")
    val id: String,

    @ColumnInfo(name = "shelf_id")
    val shelfId: String,

    @ColumnInfo(name = "name")
    val name: String,

    @ColumnInfo(name = "created_at")
    val createdAt: String,

    @ColumnInfo(name = "updated_at")
    val updatedAt: String,

    @ColumnInfo(name = "dirty", defaultValue = "1")
    val dirty: Boolean = true,

    @ColumnInfo(name = "deleted_at")
    val deletedAt: String? = null,
)
